go/host/db/blocks.go

```diff
@@ -0,0 +1,107 @@
+package db
+
+import (
+	"bytes"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/obscuronet/go-obscuro/go/common/log"
+)
+
+// DB methods relating to batches.
+
+// headerKey = blockHeaderPrefix  + hash
+func blockHeaderKey(hash gethcommon.Hash) []byte {
+	return append(blockHeaderPrefix, hash.Bytes()...)
+}
+
+// GetHeadBlockHeader returns the block header of the current head block, or (nil, false) if no such header is found.
+func (db *DB) GetHeadBlockHeader() (*types.Header, bool) {
+	head := db.readHeadBlock(db.kvStore)
+	if head == nil {
+		return nil, false
+	}
+	return db.readBlockHeader(db.kvStore, *head)
+}
+
+// GetBlockHeader returns the block header given the hash, or (nil, false) if no such header is found.
+func (db *DB) GetBlockHeader(hash gethcommon.Hash) (*types.Header, bool) {
+	return db.readBlockHeader(db.kvStore, hash)
+}
+
+// AddBlockHeader adds a types.Header to the known headers
+func (db *DB) AddBlockHeader(header *types.Header) {
+	b := db.kvStore.NewBatch()
+	db.writeBlockHeader(b, header)
+
+	// update the head if the new height is greater than the existing one
+	headBlockHeader, found := db.GetHeadBlockHeader()
+	if !found || headBlockHeader.Number.Int64() <= header.Number.Int64() {
+		db.writeHeadBlockHash(b, header.Hash())
+	}
+
+	if err := b.Write(); err != nil {
+		db.logger.Crit("Could not write rollup.", log.ErrKey, err)
```

Comment:
I don't think we should swallow errors.

Reply:
Not swallowed - Crit() causes the program to exit.

Reply:
Raised https://github.com/obscuronet/obscuro-internal/issues/1195. Will tackle after.